app/forms/reset_password_form.rb

```diff
@@ -42,7 +42,13 @@ def handle_valid_password
 
   def update_user
     attributes = { password: password }
-    attributes[:confirmed_at] = Time.zone.now unless user.confirmed?
+
+    unless user.confirmed?
+      now = Time.zone.now
+      user.email_addresses.update(confirmed_at: now)
```

Comment:
Does this update _all_ of the users email addresses with a confirmed at date, including unconfirmed ones? That may be a problem. A user could add an unconfirmed email, send a password reset to a confirmed email, then confirm all of their email addresses including the unconfirmed one by resetting their password.

I'm not sure what impact bumping the `confirmed_at` on the email address record has. Since we aren't doing it now it seems like it may be safe to leave off if we can't easily get to the email the reset password token was sent to.

Reply:
This is a part I'm trying to understand better, but I believe this is when you have an unconfirmed account, reset your password and then set the password, with the goal here being to set the account as confirmed.

This is stuff that would have previously been handled by the `EmailAddressCallback` [here](https://github.com/18F/identity-idp/blob/7db2cfb/app/models/concerns/email_address_callback.rb#L25) by setting `confirmed_at` on the user and setting the same value on the first email address.

The window is narrow, but I think we should probably only set it on the first email address (rather than all) like the callback and do all of this in a transaction to ensure things don't get weird in the time between checking the user has no confirmed emails in `user.confirmed?` and updating the `confirmed_at`.

Reply:
I switched to that in https://github.com/18F/identity-idp/pull/6474/commits/fd2c31092ba6eb434d1401d13b893f963d76c3a7